### PR TITLE
[tests-only][full-ci]Added scenarios for quota exceed with API

### PR DESCRIPTION
**Pull request by SagarGi**
### Description
This PR adds the scenario for creation for the space when `OCIS_SPACES_MAX_QUOTA` is set and when space is created with default quota than the `OCIS_SPACES_MAX_QUOTA`.

### Related Issue:
 https://github.com/owncloud/ocis/issues/6231

### CODE REVIEW

**Comment by ownclouders** [comment]
E2E tests failed: https://drone.owncloud.com/owncloud/ocis/22956/90/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/ocis/22956/tracing/public-link-for-space-alice-2023-6-6-06-41-21.zip 


</p></details>

File: tests/acceptance/features/apiSpaces/quota.feature [hunk 1]
```diff
@@ -130,3 +130,11 @@ Feature: State of the quota
     And user "Brian" has been created with default attributes and without skeleton files
     When user "Brian" uploads file with content "more than 10 bytes content" to "lorem.txt" using the WebDAV API
     Then the HTTP status code should be "507"
+
+  @env-config
+  Scenario: create a space by setting OCIS spaces max quota
```
Comment (phil-davis) [inline tests/acceptance/features/apiSpaces/quota.feature:135]:
```suggestion
  Scenario: try to create a space with quota greater than OCIS spaces max quota
```

File: tests/acceptance/features/apiSpaces/quota.feature [hunk 2]
```diff
@@ -130,3 +130,11 @@ Feature: State of the quota
...
+    Given the config "OCIS_SPACES_MAX_QUOTA" has been set to "50"
+    And user "Brian" has been created with default attributes and without skeleton files
+    And the administrator has given "Brian" the role "Space Admin" using the settings api
+    When user "Brian" creates a space "new space" of type "project" with quota "51" using the Graph API
+    Then the HTTP status code should be "500"
```
[the remark this answers is not in the document]
Comment (SagarGi) [reply tests/acceptance/features/apiSpaces/quota.feature:140]: May be this goes to block until if it is decided to add to the expected to failure.

**Comment by SagarGi** [comment]
This PR is blocked by the issue : https://github.com/owncloud/ocis/issues/6414

**Comment by phil-davis** [comment]
> This PR is blocked by the issue : #6414

And the requirement needs to be agreed - should this "create space" request return `400` or `507` - see comment:
https://github.com/owncloud/ocis/issues/6414#issuecomment-1578173160

**Comment by phil-davis** [comment]
@ScharfViktor if someone can define the requirement - HTTP status `400` or `507` - then we could make the test expect the required response, add the one failing scenario to expected-failures and merge this PR.

The next time the tests are updated in reva, this test scenario will start running (and fail, and be added to expected failures in reva).

Then it will be "easy" for the developer. They can try to adjust the code and make the test scenario pass. That way they will know already in reva that their fix works.

**Comment by ScharfViktor** [comment]
> @ScharfViktor if someone can define the requirement - HTTP status `400` or `507` - then we could make the test expect the required response, add the one failing scenario to expected-failures and merge this PR.
> 
> The next time the tests are updated in reva, this test scenario will start running (and fail, and be added to expected failures in reva).
> 
> Then it will be "easy" for the developer. They can try to adjust the code and make the test scenario pass. That way they will know already in reva that their fix works.

I see that it should be fixed here https://github.com/owncloud/ocis/pull/6469 with status code `400`
Yes, you're right about "easy" for the developer. We wanted to reduce the list of expected failures, so we did not put new tests on the list but waited fix instead. I suggest that we wait a little.

**Review by ScharfViktor — APPROVED** [review]

**Comment by ScharfViktor** [comment]
https://github.com/owncloud/ocis/pull/6469 is done. 
I rebased and changed status code to `400`

File: tests/acceptance/features/apiSpaces/quota.feature [hunk 3]
```diff
@@ -130,3 +130,11 @@ Feature: State of the quota
...
     When user "Brian" uploads file with content "more than 10 bytes content" to "lorem.txt" using the WebDAV API
     Then the HTTP status code should be "507"
+
+  @env-config
+  Scenario: try to create a space with quota greater than OCIS spaces max quota
+    Given the config "OCIS_SPACES_MAX_QUOTA" has been set to "50"
+    And user "Brian" has been created with default attributes and without skeleton files
+    And the administrator has given "Brian" the role "Space Admin" using the settings api
+    When user "Brian" creates a space "new space" of type "project" with quota "51" using the Graph API
+    Then the HTTP status code should be "400"
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/quota.feature:140]: just to be sure
```suggestion
    When user "Brian" tries to create a space "new space" of type "project" with quota "51" using the Graph API
    Then the HTTP status code should be "400"
    And the user "Brian" should not have a space called "new space"
```

**Comment by SagarGi** [comment]
> #6469 is done. I rebased and changed status code to `400`

Thanks !!

**Review by phil-davis — APPROVED** [review]

**Review by saw-jan — APPROVED** [review]

**Pull request merged**